src/mavm.rs

```diff
@@ -977,27 +980,42 @@ pub enum AVMOpcode {
 }
 
 impl Opcode {
-    pub fn is_pure(&self) -> bool {
+    pub fn is_view(&self, _: &TypeTree) -> bool {
+        match self {
+            Opcode::AVMOpcode(AVMOpcode::Inbox)
+            | Opcode::AVMOpcode(AVMOpcode::InboxPeek)
+            | Opcode::AVMOpcode(AVMOpcode::Send)        // 2 sends can't be reordered even though no state is read
```

Comment:
Shouldn't the send opcode be write but not view?  (The comment about reordering doesn't have implications for view, because it's safe to reorder two view operations.  But write operations can't be reordered.

Reply:
You're right, it's just `write`. I'm changing this